### ISSUE TITLE
stasm metadata are probably overwritten in -enrollAll mode

### ISSUE THREAD

**Issue by koenokatachi** [biometrics/openbr#431]
I want to run Stasm over a picture with multiple faces and outline rectangles and landmarks. So

```
br -algorithm 'Open+Cascade(FrontalFace)+Stasm+Draw(inPlace=true)' -enrollAll -enroll input.png output.png
```
Stasm landmarks are only outlined for a single face.

Sample input and output:
![home2](https://cloud.githubusercontent.com/assets/11669814/10270715/04260684-6b03-11e5-9b75-721742eb130a.png)
![test](https://cloud.githubusercontent.com/assets/11669814/10270717/0aa2ef22-6b03-11e5-827b-1b1c73231e18.png)

To be sure stasm actually sees the rest of the faces I've made the outlined face not eligible for detectors. Another pair of input and output:
![home3](https://cloud.githubusercontent.com/assets/11669814/10270724/3ec93220-6b03-11e5-9969-7ba45714683b.png)
![test3](https://cloud.githubusercontent.com/assets/11669814/10270725/43c83ed8-6b03-11e5-9bc9-eafd933cde78.png)

Enrolling into ```xml``` format i see the same Points and StasmXEye entries for each face.

It looks like stasm overwrites its metadata with each invocation.

**Comment by jklontz** [comment]
Hmm yeah this is a nasty bug. @bklare @sklum Looks like the `Cascade(FrontalFace)+Stasm` combination isn't well supported currently. Without eye coordinates (searchPinned=false) STASM will re-run face detection, effectively clobbering previous face detections results and detecting the same face 3 times over in the above case. Not sure about the best way to proceed...

**Comment by sklum** [comment]
It's a known issue that's been brought up on the mailing list - if someone is willing to dig through the STASM code to remove its internal detection I'd be happy to review the PR, but the STASM source is not fun to play with.

**Comment by jklontz** [comment]
What about the alternative of removing `Cascade(FrontalFace)` from the algorithm string and detecting all faces from the start with Stasm?

**Comment by sklum** [comment]
Oh, I see. This is an issue with the released model.  Yikes.  Let me verify that that even works as expected.

**Comment by sklum** [comment]
Doesn't currently work this way, but I might be able to patch it so that it does work.

**Comment by bklare** [comment]
Inefficiency from the double call to face detection aside, I do not
understand why this issue is occurring. When using enrollAll=true, after
the call to CascadeTransform, shouldn't all the detected faces be cropped
out and processed as separate templates?

**Comment by jklontz** [comment]
CascadeTransform just sets the template metadata to store the face detection bounding boxes, it doesn't actually crop the images.

**Comment by kashyap2690** [comment]
StasmTransform can use that already detected FrontalFace QRectF and pass only that Mat to stasm. Probably with extra padding.

**Comment by jklontz** [comment]
This should be fixed by #435
